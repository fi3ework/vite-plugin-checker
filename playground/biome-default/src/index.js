const a = 'hello'
var b = 'world'
const c = '!'
var d = a + b + c
