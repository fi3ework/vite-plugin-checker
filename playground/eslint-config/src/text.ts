export const text = 'Vanilla JS/TS'
