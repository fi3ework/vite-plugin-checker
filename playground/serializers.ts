import os from 'os'

const winNewLineReg = /\/r\/n/gim
const winSepReg = /\\/g

function doesUseDoubleSlashAsPath(val: string) {
  return val.includes('//vite-plugin-checker//')
}

export const normalizeWindowsLogSerializer = {
  print(val: string, serialize) {
    let result = val
    if (os.platform() === 'win32') {
      result = result.replaceAll(winNewLineReg, '/n')
      result = result.replaceAll(process.cwd().replace(winSepReg, '/'), '<PROJECT_ROOT>')

      if (doesUseDoubleSlashAsPath(result)) {
        result = result.replaceAll(
          `//a//vite-plugin-checker//vite-plugin-checker//playground-temp`,
          '<PROJECT_ROOT>/playground-temp'
        )
        result = result.split('//').join('/')
      }
    }

    return serialize(result)
  },
  test(val) {
    if (typeof val !== 'string') return false

    if (
      (os.platform() === 'win32' &&
        (val.includes(process.cwd().replace(winSepReg, '/')) || winNewLineReg.test(val))) ||
      doesUseDoubleSlashAsPath(val)
    ) {
      return true
    }

    return false
  },
}
