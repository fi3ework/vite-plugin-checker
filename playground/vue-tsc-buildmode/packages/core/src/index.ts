import { formatMessage, capitalize, processData as utilsProcessData } from '../../utils/src/helpers';

export function processData(data: string): string {
  const processed = utilsProcessData(data);
  return formatMessage(capitalize(processed));
}