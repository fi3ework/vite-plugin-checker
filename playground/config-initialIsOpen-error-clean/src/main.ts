import { text } from './text'

const hello = 'Hello'

const rootDom = document.querySelector('#root') as HTMLElement
rootDom.innerHTML = hello + text

export {}
