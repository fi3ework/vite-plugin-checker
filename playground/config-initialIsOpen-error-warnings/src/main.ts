import { text } from './text'

const rootDom = document.querySelector('#root')! as HTMLElement
rootDom.innerHTML = text

export {}
