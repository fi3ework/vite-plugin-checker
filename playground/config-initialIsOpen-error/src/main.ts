import { text } from './text'

var hello = 'Hello'

const rootDom = document.querySelector('#root')! as HTMLElement
rootDom.innerHTML = hello + text

export {}
