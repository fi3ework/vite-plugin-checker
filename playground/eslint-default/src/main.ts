import { text } from './text'

let count: string = 0
var hello = 'Hello'

const rootDom = document.querySelector('#root')!
rootDom.innerHTML = hello + text

export {}
