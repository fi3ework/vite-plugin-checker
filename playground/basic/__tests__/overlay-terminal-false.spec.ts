/* eslint-disable max-nested-callbacks */
import stringify from 'fast-json-stable-stringify'

import {
  killServer,
  preTest,
  proxyConsoleInTest,
  sleepForServerReady,
  stripedLog,
  viteServe,
} from 'vite-plugin-checker/__tests__/e2e/Sandbox/Sandbox'
import {
  editFile,
  sleep,
  testDir,
  WORKER_CLEAN_TIMEOUT,
} from 'vite-plugin-checker/__tests__/e2e/testUtils'
import { WS_CHECKER_ERROR_EVENT } from 'vite-plugin-checker/src/client'
import { copyCode } from '../../../scripts/jestSetupFilesAfterEnv'
import { serializers } from '../../../scripts/serializers'

beforeAll(async () => {
  await preTest()
})

expect.addSnapshotSerializer(serializers)

afterAll(async () => {
  await sleep(WORKER_CLEAN_TIMEOUT)
})

describe('overlay-terminal-false', () => {
  beforeEach(async () => {
    await copyCode()
  })

  describe('serve', () => {
    afterEach(async () => {
      await killServer()
    })

    it('get initial error and subsequent error', async () => {
      let diagnostics: any
      editFile('vite.config.ts', (code) => code.replace(`// edit-slot`, `terminal: false,`))
      await viteServe({
        cwd: testDir,
        proxyConsole: () => proxyConsoleInTest(true),
        wsSend: (_payload) => {
          if (_payload.type === 'custom' && _payload.event === WS_CHECKER_ERROR_EVENT) {
            diagnostics = _payload.data.diagnostics
          }
        },
      })
      await sleepForServerReady()
      expect(stringify(diagnostics)).toMatchSnapshot()
      expect(stripedLog).toMatchSnapshot()
    })
  })
})
