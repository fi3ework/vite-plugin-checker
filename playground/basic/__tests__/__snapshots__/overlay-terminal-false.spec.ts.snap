// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`overlay-terminal-false serve get initial error and subsequent error 1`] = `"[{\\"checkerId\\":\\"ESLint\\",\\"frame\\":\\"    1 | import { text } from './text'/n    2 |/n  > 3 | var hello = 'Hello'/n      | ^^^^^^^^^^^^^^^^^^^/n    4 |/n    5 | const rootDom = document.querySelector('#root')! as HTMLElement/n    6 | rootDom.innerHTML = hello + text\\",\\"id\\":\\"<PROJECT_ROOT>/temp/basic/src/main.ts\\",\\"level\\":1,\\"loc\\":{\\"column\\":1,\\"file\\":\\"<PROJECT_ROOT>/temp/basic/src/main.ts\\",\\"line\\":3},\\"message\\":\\"Unexpected var, use let or const instead.\\",\\"stack\\":\\"\\"},{\\"checkerId\\":\\"ESLint\\",\\"frame\\":\\"    3 | var hello = 'Hello'/n    4 |/n  > 5 | const rootDom = document.querySelector('#root')! as HTMLElement/n      |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^/n    6 | rootDom.innerHTML = hello + text/n    7 |/n    8 | export {}\\",\\"id\\":\\"<PROJECT_ROOT>/temp/basic/src/main.ts\\",\\"level\\":0,\\"loc\\":{\\"column\\":17,\\"file\\":\\"<PROJECT_ROOT>/temp/basic/src/main.ts\\",\\"line\\":5},\\"message\\":\\"Forbidden non-null assertion.\\",\\"stack\\":\\"\\"}]"`;

exports[`overlay-terminal-false serve get initial error and subsequent error 2`] = `""`;
